pyretailscience/cross_shop.py

```diff
@@ -51,7 +57,7 @@ def __init__(
             msg = f"The following columns are required but missing: {missing_cols}"
             raise ValueError(msg)
 
```

Comment:
Can you add a check to ensure that if group_3_col is not none then group_3_val should not be none. And vice versa. Eg
```suggestion
if (group_3_col is None) != (group_3_val is None):
   raise ValueError("If group_3_col or group_3_val is populated, then the other must be as well")
```